### PR TITLE
Improving the Design chapter

### PR DESCRIPTION
**Pull request by funkyfuture**
this should be merged into the main `branch` before the FAQ sections is added.

### CODE REVIEW

**Review by JKatzwinkel — APPROVED** [review]
Can't find any issues apart from not understanding what *Gestaltung* means in this context, and I also feel like there should maybe be a link to the design chapter in `index.rst` so that the reader proceeds to here first rather than being confronted with the quite heavy-weight web application serving the *getting started* guide, so I'm gonna push a small commit.

File: docs/design.rst [hunk 1]
```diff
@@ -1,3 +1,5 @@
+.. _design:
```
Comment (funkyfuture) [inline docs/design.rst:1]: i can't recall the proper role (maybe also `ref`), but you don't need to explicitly create an anchor inn order to reference documents or sections.

File: docs/index.rst [hunk 2]
```diff
@@ -13,3 +13,6 @@
    glossary
    genindex
    license
+
+For more in-depth reasoning about ``delb``'s philosophy, proceed to the
```
Comment (funkyfuture) [inline docs/index.rst:17]: what about reordering the toc and making `design` the first item?

File: docs/design.rst [hunk 3]
```diff
@@ -52,21 +52,23 @@ least these two share significant design aspects of Java APIs which is perceived
 as weird and clumsy in Python code.
 lxml is a wrapper around libxml2_ which was developed by the GNOME_ developers
 for other data than text documents. Data that is strictly structured and
-expectable. Text documents are different in these regards as the encoding mixes
-different abstracted encapsulations of logical and physical text fragments. And
-they are formulated and structured for human consumers, and often printing
-devices.
+expectable. Text documents are different in these regards as natural languages
+and Gestaltung allow and lead to unprecedented manifestations for which an
```
[the remark this answers is not in the document]
Comment (funkyfuture) [reply docs/design.rst:58]: see https://de.wikipedia.org/wiki/Gestaltung , afaik there's no english term for that. yes, it is Distinktionsjargon.

my point is that *layout* or *design* would be too narrow for the general contingency of *Textträgergestaltung*.

i think that's why Sachverhalte are often spelled out in english, but that is rather my ressentiment. anyway, that's an alternative.

**Comment by funkyfuture** [comment]
lol, yeah sorry, i think there's even black for rst-snippets now.

that talk was shortly after the failed coup, wasn't it?

**Comment by JKatzwinkel** [comment]
sry for the rant in cecc342, but I suffer severe OCD every time I encounter this stylistic choice, and hence crave the sympathy and affirmation of like-minded peers.

the talk was before the coup attempt, and it even says so on the background panel on stage, so I guess Henney and Stallman are off the hook on this one... I only found that video long after anyways, but it has held a special place in my heart since then because it tells me my opinions are good `:3`

**Comment by JKatzwinkel** [comment]
should the design chapter have the caption *Design philosophy*?

File: docs/design.rst [hunk 4]
```diff
@@ -97,14 +99,16 @@ concatenate broken words would need a function that removes the element objects
...
 .. code-block:: python
 
-   def remove_elements(*elements: etree.ElementBase, keep_children=False,
-                       preserve_text=False,
-                       preserve_tail=False) -> None:
+   def remove_elements(
+       *elements: etree.ElementBase,
+       keep_children=False,
+       preserve_text=False,
+       preserve_tail=False
```
Comment (funkyfuture) [inline docs/design.rst:110]:
```suggestion
```

these do not apply with delb's data model.

File: docs/design.rst [hunk 5]
```diff
@@ -97,14 +99,16 @@ concatenate broken words would need a function that removes the element objects
...
-of more specific functions:
+Here's a working implementation from the inxs_ library for that data model:
 
 .. code-block:: python
 
-   def remove_elements(*elements: etree.ElementBase, keep_children=False,
-                       preserve_text=False,
-                       preserve_tail=False) -> None:
+   def remove_elements(
+       *elements: etree.ElementBase,
```
Comment (funkyfuture) [inline docs/design.rst:107]:
```suggestion
       *nodes: delb.NodeBase,
```

**Comment by funkyfuture** [comment]
> should the design chapter have the caption _Design philosophy_?

i guess [so](https://www.youtube.com/watch?v=DlUyHogoyvE).

**Comment by funkyfuture** [comment]
> the talk was before the coup attempt, and it even says so on the background panel on stage

nah, there's no "stay tuned, we're still planning…", but i'm impressed that a technical university bears an angel bee as mascot.

File: docs/design.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (JKatzwinkel) [reply docs/design.rst:56]: I find this much easier to understand.

File: docs/design.rst (hunk 5 above)
Comment (JKatzwinkel) [reply docs/design.rst:107]: isn't this the delbless example!?

Comment (funkyfuture) [reply docs/design.rst:107]: no, no more at this point. it's about the same task with the same data type, but using a different data model.

Comment (funkyfuture) [reply docs/design.rst]: oh no, this sparse-context view git me again. sorry, fixing it.

**Comment by funkyfuture** [comment]
a'ight, i removed my mess, squashed all together and the only open question that remains is whether i mistook your proposal as teasing me over my jargon around the question of Gestaltung.

**Pull request merged**